src/librustc_mir/interpret/memory.rs

```diff
@@ -839,16 +841,14 @@ impl<'a, 'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'a, 'mir, 'tcx, M> {
         }
 
         // Forget all the relocations.
-        for k in keys {
-            alloc.relocations.remove(&k);
-        }
+        alloc.relocations.remove_range(first ..= last);
```

Comment:
I don't think this is doing what it's supposed to be doing. Writing to the first element of a `(&(), &())`

```
00 00 00 00  00 00 00 00
|-----4----||-----5-----|
```

will result in

```
00 00 00 00  00 00 00 00
|-----9----|
```

I have a fix incoming